File: backend/src/schemas/contato.py
from pydantic import BaseModel, ConfigDict, EmailStr
from typing import Optional

#
class ContatoBase(BaseModel):
    id_cliente: int
    nome: str
    cargo: Optional[str] = None
    papel: str
    telefone: Optional[str] = None
    email: Optional[EmailStr] = None
    is_principal: bool = False
    status: str = "ativo"
    observacoes_gerais: Optional[str] = None

class ContatoCreate(ContatoBase):
    pass

class ContatoRead(ContatoBase):
    id_contato: int
    
    model_config = ConfigDict(from_attributes=True)

class ContatoUpdate(BaseModel):
    nome: str | None = None
    cargo: str | None = None
    papel: str | None = None
    telefone: str | None = None
    email: str | None = None
    is_principal: bool | None = None
    status: str | None = None
    observacoes_gerais: str | None = None

File: frontend/src/domain/contato.ts
export type Contato = {
  id: string
  clienteId: string

  nome: string
  cargo?: string
  papel: string

  telefone_whatsapp?: string
  email?: string
  isPrincipal?: boolean
  status?: 'ativo' | 'inativo' | 'arquivado'
  observacoes_gerais?: string
}
